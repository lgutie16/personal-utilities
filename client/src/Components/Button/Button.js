import React from 'react'

class Button extends React.Component {
    constructor(props){
        super(props)
        this.state = {
            value: props.value
        }

        this.toggleText = this.toggleText.bind(this)
    }

    toggleText() {
        this.setState({value: "other text"})
    }
    render(){
        return <button name="button-click" value={this.state.value} onClick={this.toggleText}>{this.state.value}</button>
    }
}

export default Button