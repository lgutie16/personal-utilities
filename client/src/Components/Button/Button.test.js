import React from 'react'
import ReactDOM from 'react-dom'
import Button from './Button'


test('Button should render ', ()=>{
   const div = document.createElement('div')
   ReactDOM.render(<Button />, div)
})

test('Button show right text', () => {
    const  wrapper = shallow(
        <Button value="button text" />
    )

    //Test initial value
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.text()).toEqual('button text'); 

    //Test button after click
    wrapper.find("button").simulate('click')
    expect(wrapper.text()).toEqual('other text'); 
   
})