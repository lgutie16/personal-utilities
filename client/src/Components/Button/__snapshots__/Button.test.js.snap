// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button show right text 1`] = `
<button
  name="button-click"
  onClick={[Function]}
  value="button text"
>
  button text
</button>
`;
